saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala

```diff
@@ -435,4 +458,17 @@ object MultiModalSpRLDataModel extends DataModel {
         else
           phraseConceptToWord(x.getSegmentConcept))
   }
+
+  private def getSegmentConcepts(img: String, seg: Integer): String = {
```

Comment:
why are you using id parameters? you have the actual instances for image or segment which help to traverse the graph easily. if you have `img` instance you can get the segments through `images(img)~>imageToSegment`.